godel_surface_detection/src/segmentation/surface_segmentation.cpp

```diff
@@ -227,71 +228,63 @@ double  SurfaceSegmentation::getSearchRadius()
 }
 
 
-bool SurfaceSegmentation::setSmoothCoef(std::vector<double> &coef)
+void SurfaceSegmentation::smoothVector(const std::vector<double> &x_in,
+                                       std::vector<double> &x_out,
+                                       const std::vector<double> &smoothing_coef)
 {
-  // smoothing filters must have an odd number of coefficients
-  if(coef.size() % 2 == 1)
+  int num_coef = smoothing_coef.size();
+  if (num_coef == 0 || num_coef % 2 != 1 || x_in.size() == 0 || x_in.size() <= num_coef)
   {
-    coef_.clear();
-    num_coef_ = coef.size();
-    double sum = 0;
-    for(const auto& c : coef)
-    {
-      coef_.push_back(c);
-      sum += c;
-    }
+    if(num_coef % 2 == 0)
+      ROS_WARN_STREAM("Filter must have an odd number of coefficients. Vector not smoothed");
 
-    // set gain to be the sum of the coefficients because we need unity gain
-    gain_ = sum;
-    return(true);
-  }
-  else
-  {
-    return(false);
+    x_out = x_in;
+    return;
   }
-}
-
 
-void SurfaceSegmentation::smoothVector(std::vector<double>&x_in, std::vector<double> &x_out)
-{
+  int gain = std::accumulate(smoothing_coef.begin(), smoothing_coef.end(), 0);
```

Comment:
`gain_` was previously a `double` and is now an `int`. Was this intended behavior?